### PR TITLE
feat: improve slack message and oauth endpoint

### PR DESCRIPTION
**Pull request by nicufk**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: pkg/slacknotifier/slack.go [hunk 1]
```diff
@@ -169,11 +212,32 @@ func SendEvent(eventType *testkube.WebhookEventType, execution testkube.Executio
...
+			if err != nil {
+				return err
+			}
+		}
 	}
+
 	return nil
 }
+
+func getFailedStepsCount(steps []testkube.ExecutionStepResult) int {
```
Comment (vsukhin) [inline pkg/slacknotifier/slack.go:235]: probably, it's better to move to testkube model package

File: pkg/slacknotifier/slack_oauth.go [hunk 2]
```diff
@@ -1,52 +1,98 @@
...
-		code := c.Params("code")
+		code := c.Query("code", "")
 		if code == "" {
 			c.Status(http.StatusBadRequest)
 			_, err := c.WriteString("Code was not provided")
 			return err
 		}
+		//fmt.Printf("\n---->code is |%s|\n", code)
+
+		var slackClient = &http.Client{Timeout: 10 * time.Second}
```
Comment (vsukhin) [inline pkg/slacknotifier/slack_oauth.go:56]: we have http package, worth to reuse it

File: pkg/slacknotifier/slack_oauth.go [hunk 3]
```diff
@@ -1,52 +1,98 @@
...
+
+		req, err := http.NewRequest(http.MethodGet, slackAccessUrl, nil)
+		if err != nil {
+			c.Status(http.StatusInternalServerError)
+			_, err = c.WriteString(fmt.Sprintf("\nFailed to create request: %+v\n", err))
+			return err
```
Comment (vsukhin) [inline pkg/slacknotifier/slack_oauth.go:62]: guess, in fiber handler better to use s.Error(c, http.Status, err)

**Review by vsukhin — COMMENTED** [review]
a few comments

File: docs/slack-integration.md [hunk 4]
```diff
@@ -2,16 +2,16 @@
 
 In order to receive notifications in slack about the status of the testing process, Testkube provides integration with slack, for this to work bellow are the steps to configure it.
 
-## Create a slack bot for testkube
+## Install Testkube bot to your slack workspace
```
Comment (jfermi) [inline docs/slack-integration.md:5]: Install Testkube bot to your Slack Workspace

File: docs/slack-integration.md [hunk 5]
```diff
@@ -2,16 +2,16 @@
...
 
-Go to [https://api.slack.com/apps](https://api.slack.com/apps) and create new app within your namespace, select app to be a Bot type, add within Review Scopes to Add, add chat:write and chat:write.public then Install to Workspace and a bot token will be generated.
+Teskube bot
+<a href="https://slack.com/oauth/v2/authorize?client_id=1943550956369.3416932538629&scope=chat:write,chat:write.public,groups:read,channels:read&user_scope="><img alt="Add Testkube bot to your Slack workspace" height="40" width="139" src="https://platform.slack-edge.com/img/add_to_slack.png" srcSet="https://platform.slack-edge.com/img/add_to_slack.png 1x, https://platform.slack-edge.com/img/add_to_slack@2x.png 2x" /></a>
 
-![img.gif](img/create-slack-bot.gif)
+![img.gif](img/add-testkube-bot-to-workspace.gif)
 
-## Create a slack channel
+## Add Testkube bot to a channel
```
Comment (jfermi) [inline docs/slack-integration.md:12]: Add Testkube bot to a Channel

File: docs/slack-integration.md [hunk 6]
```diff
@@ -2,16 +2,16 @@
...
 
-Create a new slack channel or use a existing one and get the channel id that you can find in the Channel Details window at the bottom.
+![img.gif](img/add-testkube-bot-to-channel.gif)
+## Configure Testkube to use the slack bot token
```
Comment (jfermi) [inline docs/slack-integration.md:15]: Configure Testkube to Use the Slack bot Token

File: docs/slack-integration.md (hunk 4 above)
Comment (nicufk) [inline docs/slack-integration.md:5]:
```suggestion
## Install Testkube bot to your Slack Workspace
```

Comment (jfermi) [inline docs/slack-integration.md:5]:
```suggestion
## Install Testkube bot to your Slack Workspace
```

File: docs/slack-integration.md (hunk 5 above)
Comment (jfermi) [inline docs/slack-integration.md:12]:
```suggestion
## Add Testkube bot to a Channel
```

File: docs/slack-integration.md (hunk 6 above)
Comment (jfermi) [inline docs/slack-integration.md:15]:
```suggestion
## Configure Testkube to Use the Slack bot Token
```

File: pkg/slacknotifier/slack_oauth.go (hunk 3 above)
Comment (nicufk) [reply pkg/slacknotifier/slack_oauth.go:62]: @vsukhin this is a separate handler, perhaps we will want to move it to a separate service, is it worth adding it to Server?

File: pkg/slacknotifier/slack_oauth.go (hunk 2 above)
Comment (nicufk) [reply pkg/slacknotifier/slack_oauth.go:56]: @vsukhin can you point to the exact structure?

File: pkg/slacknotifier/slack.go (hunk 1 above)
Comment (nicufk) [reply pkg/slacknotifier/slack.go:235]: @vsukhin do you mean in model_execution.go?

File: pkg/slacknotifier/slack_oauth.go (hunk 3 above)
Comment (vsukhin) [reply pkg/slacknotifier/slack_oauth.go:62]: Yes, I would add it to server. Better to have all hadlers in one place

File: pkg/slacknotifier/slack_oauth.go (hunk 2 above)
Comment (vsukhin) [reply pkg/slacknotifier/slack_oauth.go:56]: github.com/kubeshop/testkube/pkg/http

File: pkg/slacknotifier/slack.go (hunk 1 above)
Comment (vsukhin) [reply pkg/slacknotifier/slack.go:235]: model_execution_result.go for example

File: pkg/slacknotifier/slack_oauth.go [hunk 7]
```diff
@@ -1,52 +1,98 @@
...
 			return err
 		}
-		code := c.Params("code")
+		code := c.Query("code", "")
 		if code == "" {
 			c.Status(http.StatusBadRequest)
 			_, err := c.WriteString("Code was not provided")
 			return err
 		}
+		//fmt.Printf("\n---->code is |%s|\n", code)
```
[the remark this answers is not in the document]
Comment (nicufk) [reply pkg/slacknotifier/slack_oauth.go]: ✅

File: pkg/slacknotifier/slack_oauth.go (hunk 2 above)
Comment (nicufk) [reply pkg/slacknotifier/slack_oauth.go]: ✅

File: pkg/slacknotifier/slack_oauth.go (hunk 3 above)
Comment (nicufk) [reply pkg/slacknotifier/slack_oauth.go]: ✅

**Review by vsukhin — APPROVED** [review]

**Pull request merged**